metrics/metrics.go

```diff
@@ -37,44 +41,62 @@ var (
 
 	// Used in fallback checks, e.g. to avoid sending "counters" with 0 values
 	metricTypes map[MetricID]MetricType
+
+	// OTel metric instrumentation
+	meter    = otel.Meter("go.opentelemetry.io/ebpf-profiler")
+	counters = map[MetricID]metric.Int64Counter{}
+	gauges   = map[MetricID]metric.Int64Gauge{}
 )
 
 func init() {
-	defs, err := GetDefinitions()
-	if err != nil {
-		panic("extracting definitions from metrics.json")
-	}
-
+	defs := GetDefinitions()
 	metricTypes = make(map[MetricID]MetricType, len(defs))
 	for _, md := range defs {
+		if md.Obsolete {
+			continue
+		}
 		metricTypes[md.ID] = md.Type
+		switch typ := md.Type; typ {
+		case MetricTypeCounter:
+			counter, err := meter.Int64Counter(md.Name,
```

Comment:
I'm using the `name` field from `metrics.json` here, another option would be to use `field` which is "namespaced" and maybe easier to parse. Both options abide by the OTel instrument naming [requirements](https://pkg.go.dev/go.opentelemetry.io/otel/metric#hdr-Instrument_Name).